jfr/src/main/java/io/netty/jfr/JfrInstrumentedResourceLeakDetector.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.jfr;
+
+import io.netty.util.ResourceLeakDetector;
+import jdk.jfr.Category;
+import jdk.jfr.Description;
+import jdk.jfr.Event;
+import jdk.jfr.Label;
+import jdk.jfr.Name;
+
+/**
+ * A {@link ResourceLeakDetector} that creates JDK Flight Recorder events for all resource leaks.
+ */
+public class JfrInstrumentedResourceLeakDetector<T> extends ResourceLeakDetector<T> {
```

Comment:
shouldn't you also implement a `ResourceLeakDetectorFactory` that makes use of your `JfrInstrumentedResourceLeakDetector` ?

Reply:
This also means this implementation could then be final and package-private

Reply:
I'll have a look at this.

Reply:
Are you suggesting something that uses another property than `io.netty.customResourceLeakDetector` ?

Reply:
actually you are right... never mind. just make it final :)

Reply:
In case people don't want more than one...